tmt/base.py

```diff
@@ -375,10 +377,101 @@ def from_spec(cls, raw: _RawRequireFmfId) -> 'RequireFmfId':  # type: ignore[ove
         return fmf_id
 
 
-_RawRequireItem = Union[str, _RawRequireFmfId]
+class _RawRequireFile(TypedDict):
+    type: Optional[str]
+    pattern: Optional[List[str]]
+
+
+@dataclasses.dataclass
+class RequireFile(
```

Comment:
Might be pointless, but, is there a difference between `library` and `file`? I'm reading the code and it seems to me there's an overlap and also they are different as well. Is `type: library` + `pattern: ...` a valid combination? My impression was `pattern` is for listing files to include in sync, what is the assumption when combined with `type: library`?

I would prefer adding a standalone `RequireBeakerlib`  class - a subclass of `RequireFile`, if it'd help anything - to capture the differences, but so far I'm looking for better understanding.